### PR TITLE
[HB-7021] Fix memory leaks occurring when fullscreen ads are shown from an Activity

### PR DESCRIPTION
**Pull request by chauduyphanvu**

### CODE REVIEW

File: GoogleBiddingAdapter/src/main/java/com/chartboost/mediation/googlebiddingadapter/GoogleBiddingAdapter.kt [hunk 1]
```diff
@@ -1123,22 +1039,214 @@ class GoogleBiddingAdapter : PartnerAdapter {
...
-            else -> ChartboostMediationError.CM_PARTNER_ERROR
+/**
+ * Callback class for interstitial ads.
+ *
+ * @param listenerRef A [WeakReference] to the [PartnerAdListener] to be notified of ad events.
+ * @param partnerAdRef A [WeakReference] to the [PartnerAd] object containing the Google Bidding ad to be shown.
+ * @param continuationRef A [WeakReference] to the [CancellableContinuation] to be resumed once the ad is shown.
+ */
+private class InterstitialAdShowCallback(
+    listenerRef: WeakReference<PartnerAdListener?>,
```
Comment (bichenwang) [inline GoogleBiddingAdapter/src/main/java/com/chartboost/mediation/googlebiddingadapter/GoogleBiddingAdapter.kt:1052]: can just write these as private val so you don't need to declare later.

File: GoogleBiddingAdapter/src/main/java/com/chartboost/mediation/googlebiddingadapter/GoogleBiddingAdapter.kt [hunk 2]
```diff
@@ -1123,22 +1039,214 @@ class GoogleBiddingAdapter : PartnerAdapter {
...
+            }
+        }
+    }
+
+    override fun onAdShowedFullScreenContent() {
+        PartnerLogController.log(SHOW_SUCCEEDED)
+
+        continuationWeakRef.get()?.let {
+            if (it.isActive) {
+                it.resume(Result.success(partnerAdWeakRef.get() ?: return))
```
Comment (bichenwang) [inline GoogleBiddingAdapter/src/main/java/com/chartboost/mediation/googlebiddingadapter/GoogleBiddingAdapter.kt:1089]: not return. we should resume with failure

File: GoogleBiddingAdapter/src/main/java/com/chartboost/mediation/googlebiddingadapter/GoogleBiddingAdapter.kt [hunk 3]
```diff
@@ -1123,22 +1039,214 @@ class GoogleBiddingAdapter : PartnerAdapter {
...
+            }
+        }
+    }
+
+    override fun onAdClicked() {
+        PartnerLogController.log(DID_CLICK)
+
+        listenerWeakRef.get()?.onPartnerAdClicked(partnerAdWeakRef.get() ?: return)
```
Comment (bichenwang) [inline GoogleBiddingAdapter/src/main/java/com/chartboost/mediation/googlebiddingadapter/GoogleBiddingAdapter.kt:1167]: also not return. we should still log. similar to a bunch of these so I'll just mark it here once.

File: GoogleBiddingAdapter/src/main/java/com/chartboost/mediation/googlebiddingadapter/GoogleBiddingAdapter.kt [hunk 4]
```diff
@@ -65,6 +67,24 @@ class GoogleBiddingAdapter : PartnerAdapter {
...
             }
+
+        /**
+         * Convert a given Google Bidding error code into a [ChartboostMediationError].
+         *
+         * @param error The Google Bidding error code as an [Int].
+         *
+         * @return The corresponding [ChartboostMediationError].
+         */
+        internal fun getChartboostMediationError(error: Int) =
```
Comment (bichenwang) [inline GoogleBiddingAdapter/src/main/java/com/chartboost/mediation/googlebiddingadapter/GoogleBiddingAdapter.kt:78]: not private? where else do we use this?

**Review by bichenwang — APPROVED** [review]
lgtm otherwise
I just wonder who is holding a strong reference to the listener and to the actual ad then 🤔

File: GoogleBiddingAdapter/src/main/java/com/chartboost/mediation/googlebiddingadapter/GoogleBiddingAdapter.kt (hunk 4 above)
Comment (chauduyphanvu) [reply GoogleBiddingAdapter/src/main/java/com/chartboost/mediation/googlebiddingadapter/GoogleBiddingAdapter.kt:78]: the callback implementations are referencing this, and they're in their own separate classes now.

**Review by cb-jpadilla — APPROVED** [review]

**Pull request merged**